File: M3Tools/Controls/ToolsContextMenu.vb
Imports System.Windows.Forms

Public Class ToolsContextMenu
	Event RefreshView()
	Event RemoveRows()
	Event SendEmails()
	Event EditPerson()

	Private Sub RemoveRowByToolStrip(sender As Object, e As EventArgs) Handles ts_Remove.Click
		Dim res = MessageBox.Show("Are you sure you want to delete this customer?", "Delete Confirmation", MessageBoxButtons.YesNo, MessageBoxIcon.Question)

		If Not res = DialogResult.Yes Then
			Return
		End If

		RaiseEvent RemoveRows()
	End Sub

	Private Sub Reload(sender As Object, e As EventArgs) Handles ts_Refresh.Click
		RaiseEvent RefreshView()
	End Sub

	Private Sub MenuClosed(sender As Object, e As EventArgs) Handles Me.Closed
		ts_Remove.Enabled = False
	End Sub

	Public Sub ToggleRemove(enable As Boolean)
		ts_Remove.Enabled = enable
	End Sub

	Public Sub ToggleSend(enable As Boolean)
		If Not ts_Send.Visible Then
			Return
		End If

		ts_Send.Enabled = enable
	End Sub

	Public Sub ToggleEdit(enable As Boolean)
		ts_Edit.Enabled = enable
	End Sub

	Private Sub EmailFunctions(sender As Object, e As EventArgs) Handles ts_Send.Click
		RaiseEvent SendEmails()
	End Sub

	Private Sub EditClicked(sender As Object, e As EventArgs) Handles ts_Edit.Click
		RaiseEvent EditPerson()
	End Sub
End Class

File: M3Tools/Controls/ToolsContextMenu.cs
using System;
using System.Windows.Forms;

namespace SPPBC.M3Tools
{
	/// <summary>
	/// 
	/// </summary>
	public partial class ToolsContextMenu
	{
		/// <summary>
		/// An event fired when the view should be refreshed
		/// </summary>
		public event EventHandler RefreshView;

		/// <summary>
		/// An event fired when an entry should be removed
		/// </summary>
		public event Events.RemoveRowsEventHandler RemoveSelected;

		/// <summary>
		/// An event fired when an email should be sent out
		/// </summary>
		public event Events.SendEmailsEventHandler SendEmails;

		/// <summary>
		/// An event fired when an entry should be editted
		/// </summary>
		public event Events.EditSelectedEventHandler EditSelected;

		/// <summary>
		/// Set whether the remove button is enabled
		/// </summary>
		public bool RemoveEnabled { set => ts_Remove.Enabled = value; }

		/// <summary>
		/// Set whether the send emails button is enabled
		/// </summary>
		public bool SendEnabled { set => ts_Send.Enabled = value; }

		/// <summary>
		/// Set whether the edit button should be enabled
		/// </summary>
		public bool EditEnabled { set => ts_Edit.Enabled = value; }

		/// <summary>
		/// 
		/// </summary>
        public ToolsContextMenu()
        {
            InitializeComponent();
        }

        private void RemoveRowByToolStrip(object sender, EventArgs e)
        {
            var res = MessageBox.Show("Are you sure you want to delete this customer?", "Delete Confirmation", MessageBoxButtons.YesNo, MessageBoxIcon.Question);

            if (!(res == DialogResult.Yes))
            {
                return;
            }

            RemoveSelected?.Invoke();
        }

        private void Reload(object sender, EventArgs e)
        {
            RefreshView?.Invoke(Parent, e);
        }

        private void MenuClosed(object sender, EventArgs e)
        {
            ts_Remove.Enabled = false;
        }


        private void EmailFunctions(object sender, EventArgs e)
        {
            SendEmails?.Invoke();
        }

        private void EditClicked(object sender, EventArgs e)
        {
            EditSelected?.Invoke();
        }
    }
}
